main.rb

```diff
@@ -1,13 +1,180 @@
-puts person.can_use_services?
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
 
-student = Student.new(5, 23, 'SaMaa')
-puts student.name
-puts student.validate_name
+  def general_person_inputs
+    print 'Age: '.cyan
+    age = gets.chomp.to_i
+    print 'Name: '.cyan
+    name = gets.chomp
+    { age: age, name: name }
+  end
 
-teacher = Teacher.new('Physics', 40, 'Taher')
-puts teacher.can_use_services?
+  def create_student
+    age, name = general_person_inputs.values_at(:age, :name)
+    print 'Parent permission (y/n): '.cyan
+    parent_permission = gets.chomp.downcase == 'y'
+    Student.new(age, name, parent_permission: parent_permission)
+  end
+
+  def create_teacher
+    age, name = general_person_inputs.values_at(:age, :name)
+    print "What is the teacher's specialization? ".cyan
+    specialization = gets.chomp
+    Teacher.new(specialization, age, name)
+  end
+
+  def create_person
+    print 'Would you like to create a student or teacher? (s/t): '.cyan
+    type = gets.chomp.downcase
+    case type
+    when 's'
+      @people << create_student
+      puts "Student \"#{@people.last.name}\" has been created successfully!".green
+    when 't'
+      @people << create_teacher
+      puts "Teacher \"#{@people.last.name}\" has been created successfully!".green
+
+    else
+      puts "Invalid input \"#{type}\", Please enter valid input (s/t).".red
+      create_person
+    end
+  end
+
+  def create_book
+    print 'Title: '.cyan
+    title = gets.chomp
+    print 'Author: '.cyan
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts "Book #{title} has been created successfully!".green
+  end
+
+  def list_books(print_index: false)
+    puts 'No books have been created yet.'.blue if @books.empty?
+
+    @books.each_with_index do |book, index|
+      print "#{index}) ".blue if print_index
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people(print_index: false)
+    puts 'No people have been created yet.'.blue if @people.empty?
+
+    @people.each_with_index do |val, index|
+      print "#{index}) ".blue if print_index
+      if val.is_a?(Student)
+        print '[Student]'
+      else
+        print '[Teacher]'
+      end
+
+      puts " ID: #{val.id}, Name: #{val.name}, Age: #{val.age}"
+    end
+  end
+
+  def ask_for_book_index
+    puts 'Please choose a book from this list by number: '.cyan
+    list_books(print_index: true)
+    gets.chomp.to_i
+  end
+
+  def ask_for_person_index
+    puts 'Please choose a person from this list by number (not ID): '.cyan
+    list_people(print_index: true)
+    gets.chomp.to_i
+  end
+
+  def create_rental
+    person_index = ask_for_person_index
+    person = @people[person_index]
+    book_index = ask_for_book_index
+    book = @books[book_index]
+
+    puts 'Please enter the rental date (dd/mm/yyyy): '.cyan
+    date = gets.chomp
+    @rentals << Rental.new(date, book, person)
```

Comment:
- [ ] It would be better if the create rental functionality works as expected :+1: Currently, we have a few bugs which needs to be fixed. Kindly check the screenshot to prevent any misunderstanding of the point raised
![Screenshot from 2022-01-28 13-19-16](https://user-images.githubusercontent.com/46388113/151554080-b6dcb301-7832-4333-9618-0f8106e49bf9.png)